Template.php

```diff
@@ -2343,4 +2352,22 @@ public function modifications ($type='all') {
   public function is_modified () {
     return (bool) count($this->modifications('modifications'));
   }
+  
+  public function isbn10Toisbn13 ($isbn10) {
+       $isbn10 = trim($isbn10);  // Remove leading and trailing spaces
+       $isbn10 = str_replace(array('—','―','–','−','‒'),'-', $isbn10); // standardize dahses : en dash, horizontal bar, em dash, minus sign, figure dash, to hyphen.
+       if (preg_match("/[^0123456789Xx-]/",$isbn10) === 1)  return $isbn10;  // Contains invalid characters
```

Comment:
Need to escape the hyphen.
`"/[^0123456789Xx\-]/"`
Or more concisely
`"/[^0-9Xx\-]/"`

For consistency I'd also suggest using `~` instead of `/` as the regular expression delimiter.

Reply:
Are you able to add a test case too, please?